thirdparty/patchmatchnet/models/net.py

```diff
@@ -0,0 +1,365 @@
+"""Implementation of PatchmatchNet
+    reference: https://github.com/FangjinhuaWang/PatchmatchNet
+
+"""
+from typing import Dict, List, Any
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from thirdparty.patchmatchnet.models.module import ConvBnReLU, depth_regression
+from thirdparty.patchmatchnet.models.patchmatch import PatchMatch
+
+
+class FeatureNet(nn.Module):
+    """Feature Extraction Network"""

```

Comment:
what is the FeatureNet used for?

Reply:
to extract features from original images (similar to feature pyramid)

Reply:
used at the beginning of patchmatchnet